### DISCUSSION TITLE
Permission denied/unable to access after new token

### DISCUSSION THREAD

**Discussion by ranabanik** [community/community#23475] [New to GitHub]
<p>My personal access token expired yesterday and I created a new one and revoked/deleted the old/expired ones. Now if I try to commit-push the repo, I get the following error message:</p>
<pre><code class="lang-auto">remote: Permission to ranabanik/spinalinjury.git denied to ranabanik.
fatal: unable to access 'https://github.com/ranabanik/spinalinjury.git/': The requested URL returned error: 403

</code></pre>
<p>What to do?</p>

**Comment by jsoref** [comment]
<p>The answer will depend on what platform you’re using… This page covers two platforms, but the answer may differ if you’re using something else:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://faq.miniorange.com/knowledgebase/reset-saved-git-login-details-from-local-machine/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/2119212/181077271-fa150cf8-37e8-4499-9984-0700ed0e268e.png" class="site-icon" width="16" height="16">

      <a href="https://faq.miniorange.com/knowledgebase/reset-saved-git-login-details-from-local-machine/" target="_blank" rel="noopener">Knowledge Base - miniOrange</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/388;"><img src="https://user-images.githubusercontent.com/2119212/181077273-e9425067-aba0-4be5-8adf-1c27d29be591.png" class="thumbnail" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7e7e03b7a7fe44cfacfdbb4b8dedb7b499194ad_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7e7e03b7a7fe44cfacfdbb4b8dedb7b499194ad_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/7/a7e7e03b7a7fe44cfacfdbb4b8dedb7b499194ad.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7e7e03b7a7fe44cfacfdbb4b8dedb7b499194ad_2_10x10.png"></div>

<h3><a href="https://faq.miniorange.com/knowledgebase/reset-saved-git-login-details-from-local-machine/" target="_blank" rel="noopener">How to reset credentials entered for Git Authentication in the local machine?...</a></h3>

  <p>Windows: Press Window button and search for Credential Manager and open it. You’ll see two tabs – Web credentials and Window Credential, select Window Credentials. Under generic credentials section, identify your git credentials, it will be in format...</p>

  <p>
    <span class="label1">Est. reading time: 1 minute</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by ranabanik** [comment]
<p>I was using Ubuntu 20.4 LTS OS.<br>
I think the link does not have the solution for that?</p>

**Comment by airtower-luna** [comment]
<p>Peculiar, if you just install the <code>git</code> package on Ubuntu there’s no credential cache active, and I hope you’d be aware if you activated one. :sweat_smile:</p>
<p>First places to check:</p>
<ul>
<li>
<code>~/.gitconfig</code>, see if you enabled any kind of credential cache.</li>
<li>
<code>.git/config</code> inside the repository, check if you configured anything there (e.g. an <code>Authorization</code> header).</li>
<li>If you push through an IDE, check the IDE config, too.</li>
</ul>

**Comment by ranabanik** [comment]
<pre><code class="lang-auto">(ims37) banikr@banikr:~/PycharmProjects/spinalinjury (msmlfunc)$ ~/.gitconfig 
/home/banikr/.gitconfig: line 1: [user]: command not found
/home/banikr/.gitconfig: line 2: name: command not found
/home/banikr/.gitconfig: line 3: email: command not found
</code></pre>
<pre><code class="lang-auto">(ims37) banikr@banikr:~/PycharmProjects/spinalinjury (msmlfunc)$ .git/config 
.git/config: line 1: [core]: command not found
.git/config: line 2: repositoryformatversion: command not found
.git/config: line 3: filemode: command not found
.git/config: line 4: bare: command not found
.git/config: line 5: logallrefupdates: command not found
.git/config: line 6: [remote: command not found
.git/config: line 7: url: command not found
.git/config: line 8: fetch: command not found
.git/config: line 9: [branch: command not found
.git/config: line 10: remote: command not found
.git/config: line 11: merge: command not found
</code></pre>
<p>I am using pycharm IDE. Not sure of any credential cache activation or so I performed. I started having issue after expiration and regenerating personal access token. (and maybe deleting/revoking old ones)</p>

**Comment by airtower-luna** [comment]
<p>The files I mentioned are configuration files to look at, not commands to run. :wink: It’s concerning that they appear to be set as executable, though.</p>
<p>I’m not familiar with Pycharm, so I can’t comment on whether it might cache anything.</p>

**Comment by ranabanik** [comment, marked as answer]
<p>Hello,<br>
I found my mistake:</p>
<p>When generating a new token, I didn’t select the <code>repo</code> option. It slipped past me somehow.</p>
<p>Thanks all for helping!</p>

**Comment by queenofcorgis** [comment]
<p>Way to self-solve <a class="mention" href="/ranabanik">@ranabanik</a> :tada: I’ve gone ahead and marked the solution for anyone else in a similar situation who encounters this post :stars:</p>
<p><a class="mention" href="/jsoref">@jsoref</a> and <a class="mention" href="/airtower-luna">@airtower-luna</a> thanks for jumping in! We appreciate you :heart:</p>
